### PR TITLE
Project structure change

### PR DESCRIPTION
**Pull request by Rashmil-1999**
This PR adds a basic setup for the front end with updated dependencies and a changed project structure directory. Since this PR adds new packages, delete the existing node modules folder and reinstall. The front end now runs on port 8080. (No particular reason why but it defaulted in the webpack file)

### CODE REVIEW

**Comment by navarroc** [comment]
App.tsx needs to be app.tsx based on the webpack and package settings.

**Comment by navarroc** [comment]
What are your thoughts on adding open: true to the webpack.dev.js? The nice thing is, when you npm start it opens the browser automatically, which is convenient.

**Comment by navarroc** [comment]
Once we get this merged I can help cleanup the docker pieces so it builds.

**Comment by navarroc** [comment]
npm run build fails. We should fix that before merging.

**Comment by Rashmil-1999** [comment]
I made the requested changes and now the build should run successfully. Also, we require node >= 16 so I update the GitHub actions to test those versions. Also, I added the redirect option in `webpack.dev.js`. Please pull this branch and let me know if everything works perfectly fine on your machine without errors or not. Thanks :D

**Comment by Rashmil-1999** [comment]
dang! the build failed let me look into it.

**Comment by Rashmil-1999** [comment]
I see that the docker build has failed. Do you want to make a pr pointing to this PR? or should we just merge this one and you want to create a separate PR for this? @navarroc

**Comment by navarroc** [comment]
> I see that the docker build has failed. Do you want to make a pr pointing to this PR? or should we just merge this one and you want to create a separate PR for this? @navarroc

Let's fix the docker in a separate PR. I can help with that while you focus on the content.

File: package.json [hunk 1]
```diff
@@ -2,51 +2,86 @@
   "name": "frontend",
   "version": "0.1.0",
   "private": true,
+  "main": "src/app.tsx",
```
Comment (navarroc) [inline package.json]: Can you fix the case on app.tsx?

Comment (Rashmil-1999) [reply package.json]: ohh, this I missed it!

**Review by navarroc — APPROVED** [review]

**Pull request merged**